### PR TITLE
missing unistd.h including for F_OK for non Linux oses.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/preferences.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 #include <sys/stat.h> // for setting the permissions of the preferences file
 #include <boost/concept_check.hpp>
+#include <unistd.h>
```
Comment (aginor) [inline src/preferences.cpp]: can you please put #ifndef _WIN32 around this? Otherwise you'll break the build on windows.

**Comment by devnexen** [comment]
good point forgot this !

**Comment by gfgtdf** [comment]
I'd reccomend to merge all these commits into one.

**Comment by devnexen** [comment]
good point, done.

**Pull request merged**